tests/spice_test.go

```diff
@@ -181,6 +192,41 @@ var _ = Describe("Vmlifecycle", func() {
 		}, 30)
 	})
 
+	Context("Migrate VM with spice connection", func() {
+
+		BeforeEach(func() {
+			if len(tests.GetReadyNodes()) < 2 {
+				Skip("To test migrations, at least two nodes need to be active")
+			}
+			// Create the VM
+			result := virtClient.RestClient().Post().Resource("virtualmachines").Namespace(tests.NamespaceTestDefault).Body(vm).Do()
+			obj, err := result.Get()
+			Expect(err).To(BeNil())
+
+			// Block until the VM is running
+			tests.WaitForSuccessfulVMStart(obj)
+		})
+
+		It("should allow accessing the spice device on the VM", func(done Done) {
+			sourceNode := getVmNode()
+
+			migration := tests.NewRandomMigrationForVm(vm)
+			err = virtClient.RestClient().Post().Resource("migrations").Namespace(tests.NamespaceTestDefault).Body(migration).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(getVmNode, 30*time.Second, time.Second).ShouldNot(Equal(sourceNode))
+
+			Eventually(func() v1.VMPhase {
+				obj, err := virtClient.RestClient().Get().Resource("virtualmachines").Namespace(tests.NamespaceTestDefault).Name(vm.GetObjectMeta().GetName()).Do().Get()
+				Expect(err).ToNot(HaveOccurred())
+				fetchedVM := obj.(*v1.VirtualMachine)
+				return fetchedVM.Status.Phase
+			}, 10*time.Second, time.Second).Should(Equal(v1.Running))
+
+			checkSpiceConnection()
+			close(done)
+		}, 30)
```

Comment:
I think you can drop the overall timeout on the test (also remove the done channel) and maybe give the VM a little bit more time than 10 seconds to really be migrated (I think we use 20 in other migration tests).